docs/io/fits/usage/cloud.rst

```diff
@@ -0,0 +1,268 @@
+.. currentmodule:: astropy.io.fits
+
+..  _fits_io_cloud:
+
+Obtaining subsets from cloud-hosted FITS files
+**********************************************
+
+Astropy offers support for extracting data from FITS files stored in the cloud.
+Specifically, the `astropy.io.fits.open` function accepts the ``use_fsspec``
+and ``fsspec_kwargs`` parameters, which allow remote files to be accessed in an
+efficient way using the `fsspec <https://filesystem-spec.readthedocs.io>`__
+package.
+
+``fsspec`` is an optional dependency of Astropy which supports reading
+files from a range of remote and distributed storage backends, such as Amazon
+and Google Cloud Storage.  This chapter explains its use.
+
+.. note::
+
+    The examples in this chapter require ``fsspec`` which is an optional
+    dependency of Astropy.  See :ref:`installing-astropy` for details on
+    installing optional dependencies.
+
+
+Subsetting FITS files hosted on an HTTP web server
+==================================================
+
+A common use case for ``fsspec`` is to read subsets of FITS data from a web
+server which supports serving partial files via the
+`Range Requests <https://en.wikipedia.org/wiki/Byte_serving>`__ feature of the
+HTTP protocol.  Most web servers support serving portions of files in this way.
+
+For example, let's assume you want to retrieve data from a large image obtained
+by the Hubble Space Telescope available at the following url::
+
+    >>> # Download link for a large Hubble archive image (213 MB)
+    >>> url = "https://mast.stsci.edu/api/v0.1/Download/file/?uri=mast:HST/product/j8pu0y010_drc.fits"
+
+This file can be opened by passing the url to `astropy.io.fits.open`.
+By default, Astropy will download the entire file to local disc before opening
+it.  This works fine for small files but tends to require a lot of time and
+memory for large files.
+
+You can improve the performance for large files by passing the parameters
+``use_fsspec=True`` and ``lazy_load_hdus=True`` to `open`.  This will make
+Astropy use ``fsspec`` to download only the necessary parts of the FITS file.
+For example:
+
+.. doctest-requires:: fsspec
```

Comment:
to be moved to the top of the file if/when https://github.com/astropy/pytest-doctestplus/issues/183 becomes available.